src/Microsoft.Identity.Web.TokenCache/MsalAbstractTokenCacheProvider.cs

```diff
@@ -98,7 +112,28 @@ private async Task OnBeforeAccessAsync(TokenCacheNotificationArgs args)
             {
                 byte[] tokenCacheBytes = await ReadCacheBytesAsync(args.SuggestedCacheKey, CreateHintsFromArgs(args)).ConfigureAwait(false);
 
-                args.TokenCache.DeserializeMsalV3(UnprotectBytes(tokenCacheBytes), shouldClearExistingCache: true);
+                try
+                {
+                    args.TokenCache.DeserializeMsalV3(UnprotectBytes(tokenCacheBytes), shouldClearExistingCache: true);
+                }
+                catch (MsalClientException exception)
+                {
+                    if (_logger != null)
+                    {
+                        Logger.CacheDeserializationError(
+                          _logger,
+                          args.SuggestedCacheKey,
+                          _protector != null,
+                          exception.Message,
+                          exception);
+                    }
+                    // Adding a better message specifically for JSON parsing error
+                    if (exception.ErrorCode == MsalError.JsonParseError)
```

Comment:
Catching just the parsing error, since this is the specific case where more info needs to be added.